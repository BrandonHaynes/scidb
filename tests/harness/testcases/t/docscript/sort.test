
--test
--start-query-logging

--set-format dcsv
--start-igdata
CREATE ARRAY winners <person:string, time:double, event:string> [year=1996:2008,1000,0]
create array event <event:string> [i=0:11,12,0]
create array year <year:int64>[i=0:11,12,0]
create array person <person:string>[i=0:11,12,0]
create array country <country:string>[i=0:11,12,0]
create array time <time:double>[i=0:11,12,0]
store(build(event,iif(i<4,'dash',iif(i<8,'steeplechase','marathon'))),event)
store(build(year,1996+(i%4 * 4)),year)
store(build(person,'[Bailey,Greene,Gatlin,Bolt,Keter,Kosgei,Kemboi,Kipruto,Thugwane,Abera,Baldini,Wanjiru]',true),person)
store(build(country,'[Canada,USA,USA,Jamaica,Kenya,Kenya,Kenya,Kenya,USA,Ethiopia,Italy,Kenya]',true),country)
store(build(time,'[(9.84),(9.87),(9.85),(9.69),(487.12),(503.17),(485.81),(490.34),(7956),(7811),(7855),(7596)]',true),time)
store(join(event,join(person,join(year,join(country,time)))),winnersFlat)
create array event_index<event:string>[event_id=0:*,5,0]
store(uniq(sort(project(winnersFlat,event)),'chunk_size=5'),event_index)
create_array(champions,<person:string,country:string,time:double> [year=1996:2008,13,0,event_id=0:3,3,0],false)
store(redimension (project(index_lookup(winnersFlat,event_index, winnersFlat.event, event_id),person,year,country,time,event_id) ,champions),champions)
--stop-igdata
scan(champions)
sort(champions,country)
sort(project(winnersFlat,country, year, event), country, year desc, 100)
--set-format dense
store(build(<val:double>[i=0:2,3,0, j=0:2,3,0],i%3+j),A)
--set-format dcsv
sort(apply(A,dim,i),dim)
--start-igdata
create array numbers <number:int64 null>[i=0:4,5,0]
--shell  --command="csv2scidb -s 1 -p N < $DOC_DATA/mostNumbers.csv > $DOC_DATA/mostNumbers.scidb "
--set-format dense
--stop-igdata
--shell --store-all --command "iquery -c $IQUERY_HOST -p $IQUERY_PORT -odense -aq "load(numbers,'${TEST_DATA_DIR}/doc/mostNumbers.scidb')""
sort(numbers, number asc)
sort(numbers, number desc)

--stop-query-logging


--cleanup

remove(winners)
remove(event)
remove(year)
remove(person)
remove(country)
remove(time)
remove(winnersFlat)
remove(event_index)
remove(champions)
remove(A)
remove(numbers)
