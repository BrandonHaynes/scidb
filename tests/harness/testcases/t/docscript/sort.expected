SCIDB QUERY : <CREATE ARRAY winners <person:string, time:double, event:string> [year=1996:2008,1000,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array event <event:string> [i=0:11,12,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array year <year:int64>[i=0:11,12,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array person <person:string>[i=0:11,12,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array country <country:string>[i=0:11,12,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array time <time:double>[i=0:11,12,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(event,iif(i<4,'dash',iif(i<8,'steeplechase','marathon'))),event)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(year,1996+(i%4 * 4)),year)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(person,'[Bailey,Greene,Gatlin,Bolt,Keter,Kosgei,Kemboi,Kipruto,Thugwane,Abera,Baldini,Wanjiru]',true),person)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(country,'[Canada,USA,USA,Jamaica,Kenya,Kenya,Kenya,Kenya,USA,Ethiopia,Italy,Kenya]',true),country)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(time,'[(9.84),(9.87),(9.85),(9.69),(487.12),(503.17),(485.81),(490.34),(7956),(7811),(7855),(7596)]',true),time)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(join(event,join(person,join(year,join(country,time)))),winnersFlat)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array event_index<event:string>[event_id=0:*,5,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(uniq(sort(project(winnersFlat,event)),'chunk_size=5'),event_index)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create_array(champions,<person:string,country:string,time:double> [year=1996:2008,13,0,event_id=0:3,3,0],false)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(redimension (project(index_lookup(winnersFlat,event_index, winnersFlat.event, event_id),person,year,country,time,event_id) ,champions),champions)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <scan(champions)>
{year,event_id} person,country,time
{1996,0} 'Bailey','Canada',9.84
{1996,1} 'Thugwane','USA',7956
{1996,2} 'Keter','Kenya',487.12
{2000,0} 'Greene','USA',9.87
{2000,1} 'Abera','Ethiopia',7811
{2000,2} 'Kosgei','Kenya',503.17
{2004,0} 'Gatlin','USA',9.85
{2004,1} 'Baldini','Italy',7855
{2004,2} 'Kemboi','Kenya',485.81
{2008,0} 'Bolt','Jamaica',9.69
{2008,1} 'Wanjiru','Kenya',7596
{2008,2} 'Kipruto','Kenya',490.34

SCIDB QUERY : <sort(champions,country)>
{n} person,country,time
{0} 'Bailey','Canada',9.84
{1} 'Abera','Ethiopia',7811
{2} 'Baldini','Italy',7855
{3} 'Bolt','Jamaica',9.69
{4} 'Keter','Kenya',487.12
{5} 'Kosgei','Kenya',503.17
{6} 'Kemboi','Kenya',485.81
{7} 'Wanjiru','Kenya',7596
{8} 'Kipruto','Kenya',490.34
{9} 'Thugwane','USA',7956
{10} 'Greene','USA',9.87
{11} 'Gatlin','USA',9.85

SCIDB QUERY : <sort(project(winnersFlat,country, year, event), country, year desc, 100)>
{n} country,year,event
{0} 'Canada',1996,'dash'
{1} 'Ethiopia',2000,'marathon'
{2} 'Italy',2004,'marathon'
{3} 'Jamaica',2008,'dash'
{4} 'Kenya',2008,'steeplechase'
{5} 'Kenya',2008,'marathon'
{6} 'Kenya',2004,'steeplechase'
{7} 'Kenya',2000,'steeplechase'
{8} 'Kenya',1996,'steeplechase'
{9} 'USA',2004,'dash'
{10} 'USA',2000,'dash'
{11} 'USA',1996,'marathon'

SCIDB QUERY : <store(build(<val:double>[i=0:2,3,0, j=0:2,3,0],i%3+j),A)>
[[(0),(1),(2)],[(1),(2),(3)],[(2),(3),(4)]]

SCIDB QUERY : <sort(apply(A,dim,i),dim)>
{n} val,dim
{0} 0,0
{1} 1,0
{2} 2,0
{3} 1,1
{4} 2,1
{5} 3,1
{6} 2,2
{7} 3,2
{8} 4,2

SCIDB QUERY : <create array numbers <number:int64 null>[i=0:4,5,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <iquery -c $IQUERY_HOST -p $IQUERY_PORT -odense -aq "load(numbers,'${TEST_DATA_DIR}/doc/mostNumbers.scidb')">
[(0),(1),(2),(null),(4)]

SCIDB QUERY : <sort(numbers, number asc)>
[(null),(0),(1),(2),(4)]

SCIDB QUERY : <sort(numbers, number desc)>
[(4),(2),(1),(0),(null)]

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

